File: Archived Problems - Project Euler/problem2.cpp
#include <iostream>
using namespace std;
void EvenFibonacciNumbers(int &FirstTerm, int &SecondTerm, int &EvenSum)
{
	FirstTerm = 1;
	SecondTerm = 2;
	EvenSum = 0;
	while (SecondTerm <= 4000000)
	{
		if (SecondTerm % 2 == 0)
		{
			EvenSum += SecondTerm;
		}
		int NextTerm = FirstTerm + SecondTerm;
		FirstTerm = SecondTerm;
		SecondTerm = NextTerm;
	}
	cout << "The sum of even-valued terms in the Fibonacci sequence up to four million is: " << EvenSum << endl;
}
int main()
{
	int FirstTerm, SecondTerm, EvenSum;
	EvenFibonacciNumbers(FirstTerm, SecondTerm, EvenSum);
	return 0;
}

File: JavaScript-Problems/Archived Problems - Project Euler/problem2.js
/* 
Problem 2 : Calculates The Sum Of The Even-Valued Term In The Fibonacci Sequence
Whose Valued Do Not Exceed Four Million 4000000
return number The Sum Of Even Fibonacci Numbers
*/

function evenFibonacciNumbers(){
    let FirstTerm = 1;
    let SecondTerm = 2;
    let EvenSum = 0;
    const MaxLimit = 4000000;

    while(SecondTerm <= MaxLimit){
        if (SecondTerm % 2 === 0){
            EvenSum += SecondTerm;
        }
        const NextTerm = FirstTerm + SecondTerm;
        FirstTerm = SecondTerm;
        SecondTerm = NextTerm;
    }
    console.log("The Sum Of The Even-Valued Term In The Fibonacci Sequence Whose Valued Do Not Exceed Four Million is :" + " " + EvenSum);
    return EvenSum;
}

evenFibonacciNumbers();